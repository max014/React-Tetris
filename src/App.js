import React from 'react';
import styles from './App.module.css';
import Display from './components/Display/Display';
import HighScores from './components/HighScores/HighScores';
import { connect } from 'react-redux';
import Modal from './components/Modal/Modal';
import Dash from './components/Dash/Dash';
import Instructions from './components/Instructions/Instructions';

export const App = props => (
	<React.Fragment>
		<div className={styles.App}>
			<HighScores className={styles.Main} />
			{props.modalUp ? <Modal /> : <Display/>}
			<div style={{display: 'flex', flexDirection: 'column', width: '300px'}}>
				<Dash />
				<Instructions />
			</div>
		</div>
		<div className={styles.TooSmall}>
			<h2>This screen is too narrow.<br/> Please use a computer.</h2>
		</div>
	</React.Fragment>
);

const mapStateToProps = state => {
 return {
	 modalUp: state.modalUp
 }
}

export default connect(mapStateToProps)(App);
