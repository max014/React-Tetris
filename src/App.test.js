import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { App } from './App';

describe('App', () => {
  describe('when `modalUp` prop is false', () => {
    const app = shallow(<App modalUp={false}/>);
    it('renders properly', () => {
      expect(toJson(app)).toMatchSnapshot();
    });
    it('renders the display', () => {
      expect(app.find('Connect(Display)').exists()).toBe(true);
    });
  });

  describe('when `modalUp` prop is true', () => {
    const app = shallow(<App modalUp={true}/>);
    it('renders properly', () => {
      expect(toJson(app)).toMatchSnapshot();
    });
    it('renders the modal', () => {
      expect(app.find('Connect(Modal)').exists()).toBe(true);
    });
  });
});
