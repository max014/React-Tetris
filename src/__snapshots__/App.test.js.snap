// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App when \`modalUp\` prop is false renders properly 1`] = `
<Fragment>
  <div
    className="App"
  >
    <Connect(HighScores)
      className="Main"
    />
    <Connect(Display) />
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "width": "300px",
        }
      }
    >
      <Connect(Dash) />
      <Instructions />
    </div>
  </div>
  <div
    className="TooSmall"
  >
    <h2>
      This screen is too narrow.
      <br />
       Please use a computer.
    </h2>
  </div>
</Fragment>
`;

exports[`App when \`modalUp\` prop is true renders properly 1`] = `
<Fragment>
  <div
    className="App"
  >
    <Connect(HighScores)
      className="Main"
    />
    <Connect(Modal) />
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "width": "300px",
        }
      }
    >
      <Connect(Dash) />
      <Instructions />
    </div>
  </div>
  <div
    className="TooSmall"
  >
    <h2>
      This screen is too narrow.
      <br />
       Please use a computer.
    </h2>
  </div>
</Fragment>
`;
